lib/portlayer/storage/volume_cache_test.go

```diff
@@ -214,6 +214,52 @@ func TestVolumeCreateGetListAndDelete(t *testing.T) {
 	}
 }
 
+// createVolumes is a test helper that creates a set of num volumes on the input volume cache and volume store.
+func createVolumes(t *testing.T, op trace.Operation, v *VolumeLookupCache, storeURL *url.URL, num int) map[string]*Volume {
+	vols := make(map[string]*Volume)
+	for i := 1; i <= num; i++ {
+		id := fmt.Sprintf("ID-%d", i)
+
+		// Write to the datastore
+		vol, err := v.VolumeCreate(op, id, storeURL, 0, nil)
+		if !assert.NoError(t, err) || !assert.NotNil(t, vol) {
+			return nil
+		}
+
+		vols[id] = vol
+	}
+
+	return vols
+}
+
+func TestAddVolumesToCache(t *testing.T) {
+	mvs1 := NewMockVolumeStore()
+	op := trace.NewOperation(context.Background(), "test")
+	v := NewVolumeLookupCache(op)
+
+	storeURL, err := util.VolumeStoreNameToURL("testStore")
+	assert.NotNil(t, storeURL)
+	storeURLStr := storeURL.String()
+	v.volumeStores[storeURLStr] = mvs1
+
+	// Create 50 volumes on the volume store.
+	vols := createVolumes(t, op, v, storeURL, 50)
+
+	// Clear the volume map after it has been filled during volume creation.
+	v.vlc = make(map[string]Volume)
+
+	err = v.addVolumesToCache(op, storeURLStr, mvs1)
+	assert.Nil(t, err)
+
+	// Check that the volume map is intact again in the cache.
+	for _, expectedVol := range vols {
+		vol, err := v.VolumeGet(op, expectedVol.ID)
+		if !assert.NoError(t, err) || !assert.Equal(t, expectedVol, vol) {
```

Comment:
No need to change it for this PR, but be aware that there is a `require` package that has the same set of tests as `assert` but will immediately fail if the test fails instead of marking the test as failed but continuing the logic.
It can make the code much more declarative as you do not have to test the return value of the assert calls and handle it.